### ISSUE TITLE
Scope of navigate-to and form-action

### ISSUE THREAD

**Issue by annevk** [w3c/webappsec-csp#510]
As per https://github.com/w3c/webappsec-csp/issues/509#issuecomment-925630075 these might have broader applicability than I would have expected.

(Though perhaps the case there is that the parent is the one setting `child.location` and not the child setting its own `window.location`, which would be more palatable.)

**Comment by ArthurSonzogni** [comment]
About "navigate-to" and "form-action", I think Chrome plumbs and enforce the CSP of the document initiator of the navigation.
https://source.chromium.org/chromium/chromium/src/+/main:content/browser/renderer_host/navigation_request.cc;l=4608

It means we enforce the CSP of the document initiator of the navigation, but the navigation can happen in any frame. That sounds like a reasonable thing, especially for form-action.

What kind of alternative should we consider? To enforce the` navigation`'s `frame`'s `current-document`'s `csp_policy` instead of the `navigation`'s `initiator`'s `csp_policy`?

**Comment by annevk** [comment]
Yeah, that's probably reasonable and yeah, that might be an alternative. It would allow you to catch someone trying to navigate you. (I haven't really thought through the implications, I initially thought the model was something else entirely.)

**Comment by alexeyst** [comment]
@annevk agree, this might be of interest; in testing current Chrome's implementation, we're able to successfully prevent navigation from the document and same-origin iframes, but not from cross-origin iframes (with `target="_top"`, which succeed in navigating the top document despite the `navigate-to` CSP).
